library/L2_HAL/sensors/environment/light.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "utility/units.hpp"
+
+namespace sjsu
+{
+class Light
+{
+ public:
+  virtual bool Initialize() const                          = 0;
+  virtual units::illuminance::lux_t GetIlluminance() const = 0;
+  virtual float GetIlluminancePercentage() const           = 0;
```

Comment:
```suggestion
  virtual float GetPercentageBrightness() const           = 0;
  virtual float GetMaxIlluminance() const           = 0;
```
I think giving the maximum illuminance would be a better as it allows the developer using this class the option of what they want to do with it, for example, maybe they are looking at the saturation value of the sensor because they have an array of them and some of them have better accuracy at brighter environments then in dark environments. Knowing the maximum lux that it can read can help to know when to switch to the other sensor.

I also think that **Brightness** sounds better then Illuminance with regards to percentage.